hase/symbex/tracer.py

```diff
@@ -153,19 +146,23 @@ def __init__(
             "call", when=angr.BP_BEFORE, action=self.concretize_indirect_calls
         )
         res = self.start_state.inspect.b(
-            "successor", when=angr.BP_AFTER, action=self.concretize_state_ip
+            "successor", when=angr.BP_AFTER, action=self.concretize_ip
         )
 
     def concretize_indirect_calls(self, state: SimState) -> None:
-        assert self.instruction_idx is not None
-        assert state.ip.symbolic or state.ip == self.trace[self.instruction_idx].ip
-        state.inspect.function_address = self.trace[self.instruction_idx].ip
+        assert self.instruction is not None
+        assert state.ip.symbolic or state.ip == self.instruction.ip
```

Comment:
That's not working.
1. __bool__ for a claripy expression will always return true instead of checking satisfiable. It should be written to `state.solver.eval(state.ip) == self.instruction.ip`
2. This will always incur assertion error when calling a hooked library function. Since we don't have PLT entry in filtered trace and only leave with 1 entry instruction, the angr step first do a call to PLT entry and causes this assertion error.